.github/workflows/ci.yml

```diff
@@ -645,6 +645,31 @@ jobs:
             ./build_tools/cmake/build_all.sh \
             "${BUILD_DIR}"
 
+  byo_llvm:
+    needs: setup
+    if: fromJson(needs.setup.outputs.should-run)
```

Comment:
> * Adds a`byo_llvm` CI job. Do we want to disable running it on PRs? It's enabled for now so we get live coverage while iterating on this PR but we can always flip that.
>    * Note: the current CI job was the first to have ccache enabled so it didn't benefit from ccache hits; subsequent ones should be cheaper (this initial one took 15 minutes).

Hmm... maybe @GMNGeoffrey can weigh in? I could see a case for running this only on postsubmit, or on a schedule. I think it's pretty similar to the `[gcc, tracing, debug]` builds in that it won't _usually_ differ from the standard configuration, but we still want to know when it breaks.

Reply:
Maybe we can submit this to start writing into the ccache so we get to learn how expensive that is.

Also, a current flurry of activity around bitcode and non-standard llvm setups makes breakage perhaps more likely than it will be in the longer term. So we can start with this enabled on PRs and revisit when it feels it's not been catching a lot of issues lately.

Reply:
SGTM, but I'm also sitting next to Geoffrey as he's working on rationalizing our CI jobs and the budgets needed to support them.

Reply:
I'm going to merge this now because I think Geoffrey likes an infra-free Friday, rather than getting roped into this now.

Reply:
Now that it's writing to ccache, latency is 8 minutes:
https://github.com/openxla/iree/actions/runs/5294769511/jobs/9584449884